src/ddmd/statementsem.d

```diff
@@ -3607,6 +3607,14 @@ else
 
     override void visit(TryCatchStatement tcs)
     {
+        //printf("TryCatchStatement.semantic()\n");
+
+        if (global.params.betterC)
```

Comment:
Could you use a gate other than betterC?

Reply:
Another layer of indirection of meaning doesn't seem to add much value here.

Reply:
Try/catch/exceptions can probably work without GC in a custom runtime implementation and are very useful for higher level code. So 👍  for an extra gate variable.

Reply:
An extra gate variable can always be added if and when it adds value. I'm generally not in favor of layers of semantic indirection otherwise, and I'm known to adding such when the value becomes clear, like in: https://github.com/dlang/dmd/pull/7296/

Reply:
This is to make it consistent with #7288. I probably should have taken up on @klickverbot's advice and rename this field to discourage use.

Reply:
This matter can be easily revisited later. "The simplest design that works" applies here.

Reply:
> This matter can be easily revisited later.

It was revisited 11 hours later.  Which calls into question why the change wasn't done here instead.